app/screens/authentication-screen/authentication-screen.tsx

```diff
@@ -0,0 +1,180 @@
+import * as React from "react"
+import { useFocusEffect } from '@react-navigation/native'
+import { Alert, Image, View } from "react-native"
+import { Button } from "react-native-elements"
+import EStyleSheet from 'react-native-extended-stylesheet'
+import { useApolloClient } from "@apollo/client"
+
+import { Screen } from "../../components/screen"
+import { color } from "../../theme"
+import { palette } from "../../theme/palette"
+import { translate } from "../../i18n"
+import KeyStoreWrapper from "../../utils/storage/secureStorage"
+import BiometricWrapper from "../../utils/biometricAuthentication"
+import { resetDataStore } from "../../utils/logout"
+import type { ScreenType } from '../../types/screen'
+
+const BitcoinBeachLogo = require("../get-started-screen/bitcoinBeach3.png")
+
+const styles = EStyleSheet.create({
+  bottom: {
+    flex: 1,
+    alignItems: "center",
+    justifyContent: "flex-end",
+    marginBottom: 36,
+    width: "100%",
+  },
+
+  button: {
+    backgroundColor: palette.white,
+    borderRadius: 24,
+  },
+
+  buttonAlternate: {
+    backgroundColor: palette.lightBlue,
+    borderRadius: 24,
+  },
+
+  buttonContainer: {
+    marginVertical: 12,
+    width: "80%",
+  },
+
+  buttonTitle: {
+    color: color.primary,
+    fontWeight: "bold",
+  },
+
+  buttonAlternateTitle: {
+    color: palette.white,
+  },
+
+  container: {
+    alignItems: "center",
+    flex: 1,
+    width: "100%",
+  },
+
+  Logo: {
+    maxHeight: "50%",
+    maxWidth: "50%",
+    marginTop: 24,
+  }
+})
+
+type Props = {
+  route: any,
+  navigation: any,
+}
+
+export const AuthenticationScreen: ScreenType = ({ route, navigation }: Props) => {
+  const client = useApolloClient()
+
+  const { screenPurpose, pin, mPinAttempts } = route.params
+
+  useFocusEffect(() => {
+    attemptAuthentication()
+  })
+
+  const attemptAuthentication = () => {
```

Comment:
async?

Reply:
I don't think `attemptAuthentication` needs to be async even though `BiometricWrapper.authenticate` is asnyc. Unless we want to await a return value `BiometricWrapper.authenticate` instead of using callbacks. Is there another reason to declare `attemptAuthentication` async?